zipkin-common/src/main/scala/com/twitter/zipkin/storage/SpanStore.scala

```diff
@@ -149,6 +149,23 @@ class InMemorySpanStore extends SpanStore with CollectAnnotationQueries {
       .toList
   }
 
+  override protected def getTraceIdsByDuration(
+    serviceName: String,
```

Comment:
OK so that basically means remove the root span condition. Makes sense.
Pull request on the way!